fastlane/lib/fastlane/actions/current_build_number.rb

```diff
@@ -0,0 +1,102 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      LATEST_BUILD_NUMBER = :LATEST_BUILD_NUMBER
+    end
+
+    class CurrentBuildNumberAction < Action
+      def self.run(params)
+        require 'spaceship'
+        credentials = CredentialsManager::AccountManager.new(user: params[:username])
+        UI.message("Login to iTunes Connect (#{params[:username]})")
+        Spaceship::Tunes.login(credentials.user, credentials.password)
+        Spaceship::Tunes.select_team
+        UI.message("Login successful")
+
+        app = Spaceship::Tunes::Application.find(params[:app_identifier])
+        build_nr = nil
+        if params[:live]
+          build_nr = app.live_version.current_build_number
+        else
+          build_nr = app.edit_version.current_build_number
+        end
+        Actions.lane_context[SharedValues::LATEST_BUILD_NUMBER] = build_nr
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Returns the current build_number of either live or edit version"
+      end
```

Comment:
Can you also provide `self.details` with more details, in particular when to use this and when to use `latest_testflight_build_number`